src/OpenTelemetry/Metrics/MeterProviderBuilderExtensions.cs

```diff
@@ -107,7 +108,20 @@ public static MeterProviderBuilder AddView(this MeterProviderBuilder meterProvid
             if (metricStreamConfiguration is ExplicitBucketHistogramConfiguration histogramConfiguration)
             {
                 // Validate histogram boundaries
-                if (histogramConfiguration.Boundaries != null && !IsSortedAndDistinct(histogramConfiguration.Boundaries))
+                if (histogramConfiguration.BucketBounds == null ||
+                    !HasValidHistogramBucketBounds(histogramConfiguration.BucketBounds))
+                {
+                    throw new ArgumentException("Histogram bounds must be in ascending order with distinct values. double.NaN is not allowed.", nameof(histogramConfiguration.BucketBounds));
+                }
+
+                // If double.NegativeInfinity or double.PositiveInfinity exist in the BucketBounds, skip the value.
+                if (histogramConfiguration.BucketBounds != null)
+                {
+                    histogramConfiguration.BucketBounds = histogramConfiguration.BucketBounds.Where(x => x != double.NegativeInfinity || x != double.PositiveInfinity).ToArray();
+                }
+
+                // If the resulting BucketBounds has no bounds, use (-Inf, +Inf) as the single bucket
+                if (histogramConfiguration.BucketBounds == null || !Enumerable.Any(histogramConfiguration.BucketBounds))
```

Comment:
A user can provide an empty `double[]` as the `Boundaries`. In this case, the SDK should use `AggregationType.HistogramSumCount`.

This should be allowed to satisfy this requirement in the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view

> - Customize the aggregation - if the default aggregation associated with the Instrument does not meet the needs of the user. For example, an HTTP client library might expose HTTP client request duration as [Histogram](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#histogram) by default, but the application developer might only want the total count of outgoing requests.